### DISCUSSION TITLE
How to create a UI for the M5Stack Core 2?

### DISCUSSION THREAD

**Discussion by Slyke** [arendst/Tasmota#14357] [Q&A] [answered]
Hello, I recently installed Tasmota onto a m5Stack Core2. For those that don't know, it's basically a microcontroller with a touch panel LCD and an I2C port on it.

After booting it up, it just displays the time on the top left on the screen. Is there any documentation or anything on how to write the code for the screen (either C or Berry Script)?

**Comment by Jason2866** [comment, marked as answer]
Please use latest development version and do the config via `Autoconfig` button in the web menu
Look in Tasmota Docu for LVGL and Berry

**Reply by codedmind** [reply]
@Jason2866 i'm trying using tasmota with m5stack tough... from what i can see is the same as core 2... but even after using autoconfig for core 2 the screen is always black... i try follow the lvgl documentation and berry but the screen is always black... maybe turned off?

How can i debug it?

Thanks

**Reply by Jason2866** [reply]
M5stack is not the same as core2. Using the setup from core2 with a M5stack will not work.

**Reply by Jason2866** [reply]
Which Display is used? Maybe this setting https://github.com/arendst/Tasmota/blob/development/tasmota/displaydesc/ILI9341_M5Stack_display.ini for the display is the needed one.

**Reply by codedmind** [reply]
Maybe I'm not clear... But it is the tough ... M5stack tough... Is like core 2 but waterproof. This one https://shop.m5stack.com/products/m5stack-tough-esp32-iot-development-board-kit

**Reply by barbudor** [reply]
Seems to be the same display controller as the Core2 : IL9342C
But not the same touch controller: CHSC6540 vs FT6336U

IL9342C SPI pin mapping seems similar but some other control pins like BL are not controlled in the same way. So possibly the Core2 driver won't work off the shelf
See https://docs.m5stack.com/en/core/tough vs https://docs.m5stack.com/en/core/core2

**Reply by sfromis** [reply]
Since about 15 hours ago, Autoconfig is also available for the M5Stack Tough, but not with support for the touch controller.

**Reply by NilzOS** [reply]
How can I disable the time display in the upper left corner ?

**Reply by sfromis** [reply]
That depends on what you did to put a time display there.... Maybe you set a `DisplayMode` ?

**Reply by albert8413** [reply]
Same problem . Trying Haspmota not working also simple lvgl buttons only shown very short then the small time and date is shown in the left upper corner and destroying the screen. In the haspmota Demo same behavior. Menu is shown shortly then every second changing date and time left and the wlan and time state of the demo on the right. It seems only First row is showing. Using latest version 13.0 with autoconfig

**Reply by sfromis** [reply]
You likely have set a debugging DisplayMode to cause that top/left clock.You probably want mode 0 instead.

**Reply by albert8413** [reply]
Very short test. Seems to be ok with Displaymode 0. Will test tomorrow. Thanks a lot

**Reply by NilzOS** [reply]
Tnx @sfromis. Works. :)
